src/ir/item.rs

```diff
@@ -537,6 +660,14 @@ impl ClangItemParser for Item {
         Self::from_ty_with_id(ItemId::next(), ty, location, parent_id, context)
     }
 
+    /// This is one of the trickiest methods you'll find (probably along with
+    /// some of the ones that handle templates in `BindgenContext`).
+    ///
+    /// This method parses a type, given the potential id of that type (if
+    /// parsing it was correct), an optional location we're scanning, which is
+    /// critical some times to obtain information, an optional parent item id,
+    /// that will, if it's `None`, become the current module id, and the
+    /// context.
```

Comment:
Hmmm... I don't particularly find this comment enlightening, but I do feel properly scared now :P